### PR TITLE
013 Script for requirement [GeneralResultCodes] INVALID_DATA wrong characters - PerformAudioPassThru

### PR DESCRIPTION
**Pull request by maria-mih**
013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua 
Testing coverage available at #643

All common functions used available at #703
@istoimenova Please review

### CODE REVIEW

**Comment by istoimenova** [comment]
@maria-mih, please update script according to clarification: "Clarification about expected resut code when invalid <audioPassThruIcon> is sent"

**Comment by istoimenova** [comment]
@maria-mih, please reference the script in TestSet.

**Comment by maria-mih** [comment]
Script needs to be updated and renamed due to clarification of expected result when invalid parameter received.

**Comment by maria-mih** [comment]
@istoimenova Script is reworked and renamed from 013_ATF_P_PerformAudioPassThru_WARNINGS_All_Params_audioPassThruIcon_invalid_whitespaces_STATIC.lua to 013_ATF_P_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_whitespaces_STATIC.lua 
Edited version is available at commit 742ac55c9f2562fd68e1b9d63b312c06d56d516f

**Comment by maria-mih** [comment]
Script is to be reworked to cover several possible scenarios with different preconditions but one expected result when request sent for all parameters and image type Static and value parameter in image structure is invalid

**Comment by maria-mih** [comment]
@istoimenova 
Please note that script is renamed from 013_ATF_P_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_whitespaces_STATIC.lua 
to
013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua 
Logic is changed so that all checks for invalid value parameter are included when all parameters of request tested and imagetype = STATIC

File: test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+-- 1.1. Request is sent with audioPassThruIcon image structure with invalid value parameter
+-- and ImageType = STATIC, all other params used in PerformAudioPassThru_request 
+-- are present and within bounds 
+--
+-- 2. Performed steps
+-- Send PerformAudioPassThru (audioPassThruIcon (invalid value parameter,ImageType = STATIC), other params) from mobile to SDL and check:
+--
+-- Expected result:
+-- SDL sends PerformAudioPassThru (INVALID_DATA, success:false) to mobile app
+
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua]: @maria-mih, please remove the empty row.

File: test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+local testCasesForPolicyTable = require('user_modules/shared_testcases/testCasesForPolicyTable')
+
+--[[ Local variables ]]
+local params_invalid_data = 
+{
+  {param_value = "   ", comment = "audioPassThruIcon_only_whitespaces"}, -- audioPassThruIcon contains only whitespaces in value parameter 
+  {param_value = "iconvalue\newlinechar", comment = "audioPassThruIcon_new_line_char"}, -- audioPassThruIcon contains new line character in value parameter
+  {param_value = "iconvalue\ttabchar", comment = "audioPassThruIcon_tab_char"}, -- audioPassThruIcon contains tab character in value parameter
+  {param_value = "", comment = "audioPassThruIcon_empty"}, -- audioPassThruIcon with empty value parameter
+  {param_value = "123", comment = "audioPassThruIcon_wrong_type"}, --audioPassThruIcon contains wrong type (integer instead of string) in value parameter
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua]: @maria-mih, in this way declaration type is again string and this is not wrong type.
You should declare as: param_value = 123

File: test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  {param_value = string.rep ("a", 65535).. ".png", comment = "audioPassThruIcon_out_upper_bound"} -- audioPassThruIcon contains ourt of upper bound value parameter
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua]: @maria-mih, typo mistake: ourt -> out

File: test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua (hunk 2 above)
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua]: @maria-mih, please add one space in --audioPassThruIcon

File: test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+--[[ General Settings for configuration ]]
+Test = require('connecttest')
+require('cardinalities')
+require('user_modules/AppTypes')
+
+--[[ Preconditions ]]
+commonFunctions:newTestCasesGroup("Preconditions")
+
+commonSteps:PutFile("Precondition_PutFile_With_Icon", "icon.png")
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua]: @maria-mih, why you use putfile in scope of check invalid mobile data?

File: test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+local commonFunctions = require('user_modules/shared_testcases/commonFunctions')
+local commonSteps = require('user_modules/shared_testcases/commonSteps')
+local commonPostconditions = require('user_modules/shared_testcases/commonPreconditions')
+local testCasesForPerformAudioPassThru = require('user_modules/shared_testcases/testCasesForPerformAudioPassThru')
+local commonTestCases = require('user_modules/shared_testcases/commonTestCases')
+local testCasesForPolicyTable = require('user_modules/shared_testcases/testCasesForPolicyTable')
+
+--[[ Local variables ]]
+local params_invalid_data = 
+{
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua]: @maria-mih, why you didn't test out of lower bound?

File: test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua (hunk 1 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua]: @istoimenova removed 26b15c22dde29708857dfc525d4115955aa75d15

File: test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua (hunk 5 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua]: @istoimenova As defined in MOBILE_API.xml for 

> <struct name="Image">
>    <param name="value" minlength="0" maxlength="65535" type="String"> 
This means that we cannot test with value lower than 0 unless the string is empty, this case is covered separately.

File: test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua (hunk 2 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua]: @istoimenova Updated 26b15c22dde29708857dfc525d4115955aa75d15

File: test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua (hunk 3 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua]: @istoimenova Updated 26b15c22dde29708857dfc525d4115955aa75d15

File: test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua (hunk 4 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua]: @istoimenova I want to make sure that even though valid icon is existing it won't be used in request. 
This script is intended to be negative check extension of SUCCESS case and I keep the SUCCESS precondition.

File: test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua (hunk 2 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua]: @istoimenova Done 26b15c22dde29708857dfc525d4115955aa75d15

**Comment by maria-mih** [comment]
@istoimenova 
Please start second review iteration for updated script with loop. 
26b15c22dde29708857dfc525d4115955aa75d15

File: test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua (hunk 1 above)
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua]: @maria-mih, ACCEPTED

File: test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua (hunk 5 above)
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua]: @maria-mih, thank you.

File: test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua (hunk 2 above)
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua]: @maria-mih, ACCEPTED

File: test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua (hunk 3 above)
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua]: @maria-mih, ACCEPTED.

File: test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua (hunk 4 above)
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/013_ATF_N_PerformAudioPassThru_INVALID_DATA_All_Params_audioPassThruIcon_STATIC.lua]: @maria-mih, thank you.

**Pull request merged**